src/main/java/com/karl/Main.java

```diff
@@ -2,12 +2,33 @@
 
 import java.io.IOException;
 import com.karl.constants.Globals;
+import com.karl.echo.EchoServer;
+import com.karl.http.HTTPServer;
 import com.karl.wrappers.WrappedServerSocket;
 
 public class Main {
   public static void main(String[] args) throws IOException {
-    WrappedServerSocket socket = new ServerSocketWrapper(Globals.PORT);
-    Server server = new Server(socket);
-    server.start();
+    int port;
+    try {
+      port = Integer.parseInt(args[0]);
+    } catch (Exception e) {
+      port = Globals.PORT;
+    }
+
+    String serverType;
+    try {
+      serverType = args[1];
+    } catch (Exception e) {
+      serverType = "http";
+    }
+
+    WrappedServerSocket socket = new ServerSocketWrapper(port);
+    if (serverType.equals("echo")) {
```

Comment:
What happens if serverType is not one of the two? 
Possible refactor to a switch statement for readability.